transactionparser/src/quality.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+var assert = require('assert');
+var BigNumber = require('bignumber.js');
+
+/*
+The quality, as stored in the last 64 bits of a directory index, is stored as
+the quotient of TakerPays/TakerGets. It uses drops (1e-6 XRP) for XRP values.
+*/
+
+function adjustQualityForXRP(quality, takerGetsCurrency, takerPaysCurrency) {
+  var numeratorShift = (takerPaysCurrency === 'XRP' ? -6 : 0);
+  var denominatorShift = (takerGetsCurrency === 'XRP' ? -6 : 0);
+  var shift = numeratorShift - denominatorShift;
+  return shift === 0 ? (new BigNumber(quality)).toString() :
```

Comment:
Yeah, this was a feature request that I made back in January: https://github.com/MikeMcl/bignumber.js/issues/49